pipelinewise/fastsync/commons/utils.py

```diff
@@ -373,3 +376,40 @@ def get_pool_size(tap: Dict) -> int:
         return cpu_cores
 
     return min(fastsync_parallelism, cpu_cores)
+
+
+def gen_export_filename(tap_id: str,
+                        table: str,
+                        suffix: str = None,
+                        postfix: str = None,
+                        ext: str = None) -> str:
+    """
+    Generates a unique filename used for exported fastsync data that avoids file name collision
+
+    Default pattern:
+        pipelinewise_<tap_id>_<table>_<timestamp_with_ms>_fastsync_<random_string>.csv.gz
+
+    Args:
+        tap_id: Unique tap id
+        table: Name of the table to export
+        suffix: Generated filename suffix. Defaults to current timestamp in milliseconds
+        postfix: Generated filename postfix. Defaults to a random 8 character length string
+        ext: Filename extension. Defaults to .csv.gz
+
+    Returns:
+        Unique filename as a string
+    """
+    if not suffix:
+        suffix = datetime.datetime.now().strftime('%Y%m%d-%H%M%S-%f')
+
+    if not postfix:
+        postfix = ''.join(random.choices(string.ascii_uppercase + string.digits, k=8))
```

Comment:
We have a random string generator function that you can reuse:
https://github.com/transferwise/pipelinewise/blob/9ae492e23f3a57505d543d3116b48f7b3e95896a/pipelinewise/cli/utils.py#L517

Reply:
ooh thanks, good to know about that. Fixed by 728817d811bf5dc378230436c1bb179f67cb3e3c